app/views/editor/thang/ThangTypeEditView.coffee

```diff
@@ -337,18 +341,43 @@ module.exports = class ThangTypeEditView extends RootView
   # upload
 
   animationFileChosen: (e) ->
-    @file = e.target.files[0]
-    return unless @file
-    return unless _.string.endsWith @file.type, 'javascript'
+    file = e.target.files[0]
+    return unless file
+    return unless _.string.endsWith file.type, 'javascript'
 #    @$el.find('#upload-button').prop('disabled', true)
     @reader = new FileReader()
     @reader.onload = @onFileLoad
-    @reader.readAsText(@file)
+    @reader.readAsText(file)
+
+  animateAnimationFileChosen: (e) ->
+    file = e.target.files[0]
+    return unless file
+    if not _.string.endsWith file.type, 'javascript'
+      noty({text: "Only accepts files ending with '.js'", type:"error", timeout: 5000})
+      return
+    if not confirm("This button may have unknown effects. Are you sure you want to continue?")
+      noty({text: "Cancelled import of '.js' file", type:"info", timeout: 3000})
+      return
+    @reader = new FileReader()
+    @reader.onload = @onAnimateFileLoad
+    @reader.readAsText(file)
+
+  onAnimateFileLoad: (e) =>
+    result = @reader.result
+    parser = new AdobeAnimateParser(result)
+    parser.parse()
```

Comment:
The parser will throw errors when things fail - are those handled / surfaced to user?

Reply:
Yeah. A noty was automatically made for me when I was testing.

Reply:
Made this more explicit. Thanks